src/services/source/nativeIntegration.ts

```diff
@@ -38,9 +38,12 @@ export class NativeIntegrationSourceService implements SourceService {
     const respList: SourceTransformationResponse[] = await Promise.all<FixMe>(
       sourceEvents.map(async (sourceEvent) => {
         try {
+          const newSourceEvent = sourceEvent as { headers: any };
+          const { headers } = newSourceEvent;
+          delete newSourceEvent.headers;
           const respEvents: RudderMessage | RudderMessage[] | SourceTransformationResponse =
-            await sourceHandler.process(sourceEvent);
-          return SourcePostTransformationService.handleSuccessEventsSource(respEvents);
+            await sourceHandler.process(newSourceEvent);
+          return SourcePostTransformationService.handleSuccessEventsSource(respEvents, { headers });
```

Comment:
Do we foresee any other use case in future where headers might be needed during source transformation i.e in `process` method?

Reply:
possible but less likely for `x-` headers

Reply:
I see